src/Sylius/Component/Core/OrderProcessing/StateResolver.php

```diff
@@ -99,4 +120,15 @@ protected function getShippingState(OrderInterface $order)
 
         return OrderShippingStates::PARTIALLY_SHIPPED;
     }
+
+    /**
+     * @param StateMachineInterface $stateMachine
+     * @param string $transition
+     */
+    private function applyTransition(StateMachineInterface $stateMachine, $transition)
+    {
+        if ($stateMachine->can($transition)) {
+            $stateMachine->apply($transition);
+        }
```

Comment:
To avoid `can` we would have to add a check if current state is different than the one we would like to transit to. Then we can let the exception occur when something is wrong.